src/utils/media-utils.js

```diff
@@ -89,22 +87,16 @@ export const determineMediaDuration = memoizee(async url => {
     }
 
     try {
-      const player = React.createElement(ReactPlayer, {
-        url: validUrl,
-        light: false,
-        playing: false,
-        onDuration: durationInSeconds => {
-          const durationInMiliseconds = Math.ceil(durationInSeconds * 1000);
+      const player = React.createElement(MediaDurationIdentifier, {
+        sourceUrl: validUrl,
+        onDuration: durationInMiliseconds => {
           resolve(durationInMiliseconds);
           ensureCleanup();
-        },
-        onError: error => {
```

Comment:
If the media for some reasons cannot be loaded, we end up with an eternally unsettled Promise. I saw that for HTML5 players there was an `error` event in `plyr` (https://github.com/sampotts/plyr#standard-media-events). But it might be worth using a `Promise.race` with a timeout of 5 seconds or so to ensure there is some return value, watcha think?

Reply:
I think it'a a good idea, so much so that you have already implemented it 😋 
```
  const timeoutPromise = new Promise((_resolve, reject) => {
    setTimeout(() => {
      reject(new Error(`Timeout determining duration of ${url}`));
      ensureCleanup();
    }, MEDIA_TIMEOUT_IN_MS);
  });
  return Promise.race([playerPromise, timeoutPromise]);
  ```
  Or did you mean something else?

Reply:
Oopsie, we were inside the `playerPromise here`. We will still end up with the DOM element hanging in there and not being removed, but the user's browser will tolerate a few of them :-)